pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -272,11 +275,28 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 						CacheDuration: &durationpb.Duration{Seconds: 5 * 60},
 					},
 				}
-			} else {
+			} else if features.JwksFetchMode == jwt.Hybrid {
 				provider.JwksSourceSpecifier = push.JwtKeyResolver.BuildLocalJwks(jwtRule.JwksUri, jwtRule.Issuer, "")
+			} else {
+				// Log error and create remote JWKs with fake cluster
+				authnLog.Errorf("Failed to look up Envoy cluster %v. "+
+					"Please create ServiceEntry to register external JWKs server or "+
+					"set PILOT_JWT_ENABLE_REMOTE_JWKS to hybrid/istiod mode.", err)
+				provider.JwksSourceSpecifier = &envoy_jwt.JwtProvider_RemoteJwks{
+					RemoteJwks: &envoy_jwt.RemoteJwks{
+						HttpUri: &core.HttpUri{
+							Uri: jwtRule.JwksUri,
+							HttpUpstreamType: &core.HttpUri_Cluster{
+								Cluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", jwksInfo.Hostname, jwksInfo.Port),
```

Comment:
Does Envoy crash if this cluster is missing? What happens if the cluster is deleted first but the route is still active?

Reply:
If the cluster is not existed, Istiod will generate the remote jwks without `JwksSourceSpecifier` field, and Envoy itself will keep logging warning with this field is required.

Reply:
I think the question is the JwksSourceSpecifier is set to a cluster that is not found